tests/libc/enc/CMakeLists.txt

```diff
@@ -5,42 +5,31 @@
 # Include the libc tests wrappers:
 include(../tests.cmake)
 
-# Generate wrappers for libc tests.
-foreach(test_file ${LIBC_TESTS})
-  string(MAKE_C_IDENTIFIER ${test_file} test_name)
-  list(APPEND test_dependencies ${CMAKE_CURRENT_BINARY_DIR}/${test_file})
-  configure_file(include.c.in ${test_file})
-  string(APPEND run_test "\
-    extern int ${test_name}(int argc, const char* argv[]);\n\
-    ret += run_test(\"${test_name}\", ${test_name});\n")
-endforeach()
-
-configure_file(tests.c.in tests.c)
-
-oeedl_file(../libc.edl enclave gen)
-
 set(test_extras
-    ../../../3rdparty/musl/libc-test/src/common/vmfill.c
-    ../../../3rdparty/musl/libc-test/src/common/mtest.c
-    ../../../3rdparty/musl/libc-test/src/common/rand.c)
+    ${PROJECT_SOURCE_DIR}/3rdparty/musl/libc-test/src/common/vmfill.c
+    ${PROJECT_SOURCE_DIR}/3rdparty/musl/libc-test/src/common/mtest.c
+    ${PROJECT_SOURCE_DIR}/3rdparty/musl/libc-test/src/common/rand.c)
 
 if(NOT OE_TRUSTZONE)
     # This file requires TLS relocations, which cannot be applied by OP-TEE's
     # ELF loader.
     list(APPEND test_extras
-        ../../../3rdparty/musl/libc-test/src/functional/tls_align_dso.c)
+        ${PROJECT_SOURCE_DIR}/3rdparty/musl/libc-test/src/functional/tls_align_dso.c)
 endif()
 
-add_enclave(TARGET libc_enc UUID d7fe296a-24e9-46d1-aa78-9c7395082a41
-    SOURCES
+oeedl_file(../libc.edl enclave gen)
+
+add_library(libctest-support
     enc.c
-    ${CMAKE_CURRENT_BINARY_DIR}/tests.c
-    ${test_dependencies}
-    ${test_extras}
-    ${LIBC_TESTS}
-    ${gen})
+    ${gen}
+    ${test_extras})
 
-target_compile_options(libc_enc PRIVATE
+target_include_directories(
+    libctest-support PRIVATE
+    ${CMAKE_CURRENT_BINARY_DIR}
+    ../../../3rdparty/musl/libc-test/src/common)
```

Comment:
Replace directory traversals with `${PROJECT_SOURCE_DIR}` here and line 93 as well?

Reply:
Good point, a build toggle is probably the right way to go.